e2e/instances.spec.ts

```diff
@@ -0,0 +1,57 @@
+import { test, expect } from '@playwright/test'
+import {login, logout, uploadTestDatabase} from './utils'
+
+test.describe('new instance', () => {
+    test.beforeEach(async ({ page }) => {
+        await login(page)
+        await uploadTestDatabase(page)
+    })
+    test.afterEach(async ({ page }) => {
+        await logout(page)
+    })
+
+    test('create new dhis2 instance', async ({ page }) => {
+        test.setTimeout(10 * 60 * 1000) // 10 minutes
+
+        await page.getByRole('link', { name: 'Instances' }).click()
+        await page.getByRole('button', { name: 'New DHIS2 instance' }).click()
+
+        await expect(page.getByRole('group', { name: 'Basic information' })).toBeVisible()
+        await expect(page.getByRole('group', { name: 'Instance configuration' })).toBeVisible()
+        await expect(page.getByText('Advanced configuration')).toBeVisible()
+
+        const randomName = 'e2e-test-' + Math.random().toString().substring(8)
+        await page.getByRole('textbox', { name: 'Name'}).fill(randomName)
+        await page.getByRole('textbox', { name: 'Description'}).fill('This is an e2e test instance.')
+
+        // TODO this needs a better locator
+        await page.getByTestId('dhis2-uiwidgets-singleselectfield').filter({ hasText: 'Database' }).getByTestId('dhis2-uicore-select-input').click()
+        await page.getByText('whoami/test/empty-db.sql.gz').click()
+
+        await expect(page.getByRole('button', { name: 'Create instance' })).toBeEnabled()
+        await page.getByRole('button', { name: 'Create instance' }).click()
+        await expect(page.getByRole('cell', { name: randomName })).toBeVisible({timeout: 10000})
+
+        // TODO is there a better way to make sure instance is Running?
+        const newInstanceRow = page.locator(`//table[@data-test='dhis2-uicore-datatable']//tr[contains(td[2], '${randomName}')]`)
+
+        let status: string
+
+        do {
+            await expect(newInstanceRow.locator('td:first-child')).toHaveText(/^[A-Za-z]+$/)
+            status = await newInstanceRow.locator('td:first-child').textContent()
+            await page.waitForTimeout(5000)
+            await page.reload()
+        } while (status !== 'Running')
+
+        await expect(newInstanceRow.locator('td:first-child')).toContainText('Running')
```

Comment:
I tried something similar, but I forgot to mention one small but very important detail - it's required to reload the page in order to see a change in the status. 👀  Therefore, this doesn't really work, unless we can add a page reload in there somehow? 🤷

Reply:
Aha one of those cases where we can clean up the test after we've cleaned up the code 🙄 

First a practical suggestion, to reload the window, just call `await page.reload()` ([see docs](https://playwright.dev/docs/api/class-page#page-reload))

An on a more general note.... Since this is a single-page-application we can't really say a feature is fully functional if it requires a full page reload. So technically this is functionality is broken, and having a test in place to assert the broken functionality is working as expected but not as required feels rather odd.

Reply:
> First a practical suggestion, to reload the window, just call `await page.reload()`

Yeah, that's what I'm doing already. 👍 

> An on a more general note.... Since this is a single-page-application we can't really say a feature is fully functional if it requires a full page reload. So technically this is functionality is broken, and having a test in place to assert the broken functionality is working as expected but not as required feels rather odd.

At this point, with the instances list view getting deprecated anytime soon, I'm not sure how to continue here. Should we merge this as is and then fix the tests after the UI changes (to deployments list view) or should I wait until the new UI changes are merged and then resume working on this?

Adding @tonsV2 in the loop.

Reply:
Branches have a tendency to deprecate unless merged, so unless you really want to save this for later I'd say go ahead and merge. Just reload the page for now and let's update the test later. Please make a jira and a also comment in the code about the shortcomings.

Reply:
I have removed the instance status check for now and added a comment in the code, as well as the following Jira task to keep track of this - https://dhis2.atlassian.net/browse/DEVOPS-351